packages/template-icons/src/useUniqueId.tsx

```diff
@@ -0,0 +1,9 @@
+import { useMemo } from "react";
+
+let counter = 0;
+
+const useUniqueId = () => {
+  return useMemo(() => counter++, []);
```

Comment:
You can make this an implicit return

Reply:
Also: shouldn’t `counter` be the dependency of `useMemo`? 🤔

Reply:
Not really. Hooks are executed during render and their dependencies should come from properties and state. There is no way for react to know when to re-evaluate a hook if something outside of its control has changed.

This hook is essentially a poor man's hack to this https://github.com/facebook/react/issues/1137 . I have not found a cleaner way to do it.

Do you want me to add some comment into the code to explain it?

Reply:
What we want is a unique identifier for every "instance" of a component. That is achieved by the `useMemo`s second argument `[]` which says, execute the function and return new (incremented) value on every _mount_. And a _mount_ is when a new _"instance"_ is created.

Reply:
Does that make sense @MichaelDeBoey ?

Reply:
Maybe adding a little comment here would make this a bit clearer indeed. 🙂

Reply:
Done https://github.com/codesandbox/codesandbox-templates/pull/5/files#diff-7daac9e6713c1145ad0340ef5927b3cdR3